airship-coordinator/src/main/java/io/airlift/airship/coordinator/AwsProvisioner.java

```
@@ -428,10 +452,16 @@ String getRawAgentUserData(String instanceType, String agentConfig, List<String>
         String contentTypeText = "Content-Type: text/plain; charset=\"us-ascii\"";
         String attachmentFormat = "Content-Disposition: attachment; filename=\"%s\"";
 
-        URI partHandler = getRequiredUri(repository, new MavenCoordinates("io.airlift.airship", "airship-ec2", airshipVersion, "py", "part-handler", null));
+        Iterator<String> iterator = Splitter.on(':').split(provisioningScriptsArtifact).iterator();
```

It's better to convert this to a list using `ImmutableList.copyOf()` and verify the size, then grab the parts with `get()`. With this approach, you silently ignore extra parts and throw a `NoSuchElementException` if too few.